### PR TITLE
Mixih/backports

### PR DESCRIPTION
**Pull request by Mixih**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by Mixih** [comment]
Welp travis just disappeared from this pull....

**Comment by Mixih** [comment]
Warning: History is backwards here...... Please review and say whether this is ok

**Comment by Mixih** [comment]
Warning: History is backwards here...... Please consider that during review
CC @sils @jayvdb

**Comment by Mixih** [comment]
xref https://github.com/coala/coala/issues/3313

**Comment by jayvdb** [comment]
unack 3d87428
unack 43c09387

**Comment by jayvdb** [comment]
We dont *need* to sort requirements, or have a .travis.yml on the release branch.

**Comment by jayvdb** [comment]
`git log` and `git diff` on the release branch should only include real bug fixes.  Manual cherry-picking to remove unnecessary stuff is required.

**Comment by jayvdb** [comment]
ack 85dce47f9
ack aa18f2dac

**Comment by jayvdb** [comment]
ack f346f2d
ack 37513ce
ack 7aa92fc

**Comment by jayvdb** [comment]
ack 13f74a3

**Comment by Mixih** [comment]
History needs to be cleaned anyways so restoring wip header

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,7 @@
-PyPrint~=0.2.5
+PyPrint~=0.2.6
 Pygments~=2.1
-setuptools>=19.2
 # Do *not* use 0.3 which is not backwards compatible to common clang versions
-libclang-py3~=0.2.0
```
Comment (jayvdb) [inline requirements.txt]: IMO this shouldnt be changed.  See https://github.com/coala/coala/issues/3313#issuecomment-267873619

Comment (Mixih) [inline requirements.txt]: That's also the complete wrong version *sigh

**Comment by jayvdb** [comment]
ack 8fda27c
ack 9f17509
ack 51b73da
ack ee0c39a
ack 4b40e26
ack 82adbb5
ack e7adea8

**Comment by jayvdb** [comment]
ack 8fda27c 9f17509 51b73da ee0c39a 4b40e26 82adbb5 e7adea8

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**